source/isaaclab/isaaclab/sensors/ray_caster/ray_caster.py

```diff
@@ -250,14 +260,132 @@ def _update_buffers_impl(self, env_ids: Sequence[int]):
             ray_starts_w = quat_apply(quat_w.repeat(1, self.num_rays), self.ray_starts[env_ids])
             ray_starts_w += pos_w.unsqueeze(1)
             ray_directions_w = quat_apply(quat_w.repeat(1, self.num_rays), self.ray_directions[env_ids])
-        # ray cast and store the hits
-        # TODO: Make this work for multiple meshes?
-        self._data.ray_hits_w[env_ids] = raycast_mesh(
-            ray_starts_w,
-            ray_directions_w,
+
+        num_env = ray_starts_w.shape[0]
+        final_hits = torch.full((num_env, self.num_rays, 3), float("inf"), device=self._device)
+        mesh_ids_list = []
+        mesh_env_indices_list = []
+        # to avoid the meshes from being garbage collected we hold the references alive during the kernel execution
+        self._dynamic_wp_meshes = []
+
+        # loop over each mesh and its corresponding mesh infos
+        for pattern, mesh_info_list in self.meshes.items():
+            for mesh_info in mesh_info_list:
+                mesh_view = mesh_info["xform_view"]
+                indices = torch.arange(mesh_view.count, device=self._device)
+                # get the current world transforms for the mesh prim (for dynamic updates)
+                mesh_pos, mesh_quat = mesh_view.get_world_poses(indices)
+                base_points = torch.tensor(mesh_info["base_points"], device=self._device)
```

Comment:
I had to convert this to a float32 to make it work. It threw an error in quat_apply(). It may be worth noting it.